go/iot/protocol/opencensus/openCensusBase.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencensus
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"time"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"github.com/census-ecosystem/opencensus-experiments/go/iot/protocol"
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type OpenCensusBase struct {
+	ctx        context.Context
+	measureMap map[string]stats.Measure // Store all the measure based on their Name. Used for the future record
+	// TODO: What if different views share the same tag key
+	tagKeyMap map[string]tag.Key
+}
+
+func (census *OpenCensusBase) Initialize(projectId string, reportPeriod int) {
+	census.ctx = context.Background()
+	census.measureMap = make(map[string]stats.Measure)
+	census.tagKeyMap = make(map[string]tag.Key)
+	exporter, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID: projectId, // Google Cloud Console project ID.
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	view.RegisterExporter(exporter)
+	view.SetReportingPeriod(time.Second * time.Duration(reportPeriod))
+}
+
+func (census *OpenCensusBase) containsMeasure(name string) bool {
+	_, ok := census.measureMap[name]
+	return ok
+}
+
+// Given the censusArgument, initialize the OpenCensus framework
+func (census *OpenCensusBase) ViewRegistration(myView *(view.View)) error {
+	if census.containsMeasure(myView.Measure.Name()) == true {
```

Comment:
This check is not necessary. You could have multiple `View`s using the same `Measure`. For example define a `View` with `Count` aggregation and another with `LastValue` aggregation, and they use the same `Measure`.

Instead, you should check whether the schema of `Measure`s with the same name changed or not.

Reply:
so when one measure is bind to multiple views and I record data through that measure, would those multiple views all get that data?

Reply:
Exactly. One `Measure` can map to multiple `View`s, and recording for it will update all the corresponding `View`s. One example: https://github.com/census-instrumentation/opencensus-java/blob/master/impl_core/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java#L747-L780.

Please also fix https://github.com/census-ecosystem/opencensus-experiments/blob/master/go/iot/sensor/examples/main.go#L93-L94. You only need one `Measure` for this case.

Reply:
Could you please clarify more clearly about checking whether the schema of measures with the same name changed or not?

Reply:
For the bug you mention in the " https://github.com/census-ecosystem/opencensus-experiments/blob/master/go/iot/sensor/examples/main.go#L93-L94. You only need one Measure for this case. "I think these two measures are for two different views so that we should not delete them.

Reply:
I mean you should use one measure called ""opencensus.io/measure/sound_strength_svl_mp1_7c3c" for both views.

Measure name should not mention the aggregation type.